versioned_docs/version-2.x/advanced/analytics/server-side-tagging-tracking.mdx

```diff
@@ -0,0 +1,186 @@
+[Addingwell account](https://www.addingwell.com/)
+
+### Setup Addingwell server
+
+You must follow the official
+[Addingwell Get Started](https://docs.addingwell.com/b/2CA8BCF1-E67D-46DF-A2BE-F93017D8FD7A/Getting-started-with-Addingwell)
+documentation.
+
+### With GTM
+
+This guide will show you how to integrate Addingwell with Google Tag Manager.
+This way, the Analytics tag will be implemented with Google Tag Manager but the
+traffic will head to your custom server
+
+#### Setup GTM
+
+You must follow the
+[Addingwell Send data documentation (Option 1)](https://docs.addingwell.com/b/C45DD221-2773-4DE0-B588-39B60785291A/Send-the-data-to-your-tagging-server)
+
+#### Setup GTM Analytics plugin in Front-Commerce
+
+Install the Google Tag Manager analytics plugin
+
+```shell
+npm install @analytics/google-tag-manager
+```
+
+Then edit your analytics configuration from the `src/config/analytics.js` to add
+the Google Tag Manager plugin:
+
+```js title="src/config/analytics.js"
+export default {
+  analytics: {
+    // highlight-next-line
+    enable: true,
+    debug: process.env.NODE_ENV !== "production",
+    defaultSettings: {},
+    plugins: [
+      // highlight-start
+      {
+        name: "google-tag-manager",
+        needConsent: false,
+        enabledByDefault: true,
+        settings: (authorization) => {
+          return {
+            containerId: "GTM-XXXXXXXX", // This should be the GTM container ID of the Container that will inject Google Tag not the server container ID
+          };
+        },
+        script: () => import("@analytics/google-tag-manager"),
+      },
+      // highlight-end
+    ],
+  },
+};
+```
+
+#### Setup CSP
+
+Then you'll need to update your CSP to allow your Front-Commerce application to
+communicate with external services. Let's say that your analytics domain is
+`metrics.my-commerce.net`
+
+```js title="src/config/website.js"
+module.exports = {
+  //....
+  contentSecurityPolicy: {
+    directives: {
+      // highlight-next-line
+      scriptSrc: ["metrics.my-commerce.net", "www.googletagmanager.com"], // We need to add Google Tag Manager to allow GTM to inject GA4 tag into our pages
+      frameSrc: [],
+      styleSrc: [],
+      imgSrc: [],
+      fontSrc: [],
+      // highlight-next-line
+      connectSrc: ["metrics.my-commerce.net"],
+      baseUri: [],
+      childSrc: [],
+      defaultSrc: [],
+      manifestSrc: [],
+      mediaSrc: [],
+      objectSrc: [],
+      workerSrc: [],
+    },
+  },
+  //....
+};
+```
+
+### With GA4
+
+#### Setup your server container
+
+You first need to follow the Addingwell
+[gtag.js configuration guide (option 3)](https://docs.addingwell.com/b/C45DD221-2773-4DE0-B588-39B60785291A/Send-the-data-to-your-tagging-server).
+
+Please note your custome tagging domain we will need this for the next step (in
+the guide, we will consider that the domain is `metrics.my-commerce.net`).
+
+#### Setup GA4 plugin in Front-Commerce
```

Comment:
**nitpick:** similar to the GTM setup, I think mentioning the previous section and an updated config with only the differences highlighted would help keeping it short here